radio/src/lua/interface.cpp

```diff
@@ -144,41 +131,40 @@ void luaHook(lua_State * L, lua_Debug *ar)
 #endif // #if defined(LUA_ALLOCATOR_TRACER)
 }
 
-void luaSetInstructionsLimit(lua_State * L, int count)
+void luaEmptyEventBuffer()
 {
-  instructionsPercent = 0;
-#if defined(LUA_ALLOCATOR_TRACER)
-  lua_sethook(L, luaHook, LUA_MASKCOUNT|LUA_MASKLINE, count);
-#else
-  lua_sethook(L, luaHook, LUA_MASKCOUNT, count);
-#endif
+  events[0] = 0;
+  events[1] = 0;
 }
 
-int luaGetInputs(lua_State * L, ScriptInputsOutputs & sid)
+int luaGetInputs(lua_State * L2, ScriptInputsOutputs & sid)
```

Comment:
@jfrickmann is there a particular reason for using that cryptic `L2` parameter name?

Reply:
Yes indeed! To support preemption, Lua is called with lua_resume() instead of lua_pcall(), and L2 is the coroutine used for that. I have to use a coroutine instead of the main thread, because Lua has a "feature" that kills the thread forever if a script error happens during lua_resume(), and I can replace the coroutine but not the main thread. 
And in this function, I use both the coroutine and the main thread, because I preserve a string value by transferring it over to the main stack so that it does not get collected. The problem is that the string value otherwise would get dereferenced in Lua while an external reference is used. You can see the code from line 155:
```
        switch (field) {
          case 0:
            luaL_checktype(L2, -2, LUA_TNUMBER); // key is number
            luaL_checktype(L2, -1, LUA_TSTRING); // value is string
            lua_xmove(lsScripts, L, 1);          // To preserve the string value, move it to the main stack
            lua_pushnil(lsScripts);              // Keep the stack balanced
            lua_insert(L, -2);                   // Keep the coroutine at the top of the main stack
            si->name = lua_tostring(L, -2); // External reference to the string!
            break;
```

Reply:
Ok, so `L` is a global variable, and you changed the param name to `L2` to be able to use both at the same time, right?

Reply:
Maybe the function parameter should be more explicit, so that there is no possible confusion with the global variables?

Reply:
Yes. And we can rename the function parameter if you like to make it more clear. No problem :-)
Do you want to do it, or shall I do it? And what should we call it?

Reply:
Your PR, your choice! I'm just here for critics :-D

Reply:
Then I will use the opportunity to clean up some legacy confusion in the code :-) 
luaGetInputs and luaGetOutputs are only called from one place in the code, with L2 = lsScripts. So I will simply eliminate the L2 argument, and replace L2 with lsScripts. That should hopefully make things crystal clear!

Reply:
Please take a look now, I have cleaned it up and also added directives to exclude unused Mixer/Custom/Model script code when LUA != YES.